openmls/src/tree/tests/test_tree_truncation.rs

```diff
@@ -0,0 +1,46 @@
+use crate::{
+    ciphersuite::Ciphersuite,
+    group::{HandshakeMessageFormat, ManagedGroupCallbacks, ManagedGroupConfig, UpdatePolicy},
+    test_utils::test_framework::{ActionType, ManagedTestSetup},
+};
+
+#[test]
+fn test_truncation_after_removal() {
+    // Set up a group with 8 members.
+    let handshake_message_format = HandshakeMessageFormat::Plaintext;
+    let update_policy = UpdatePolicy::default();
+    let callbacks = ManagedGroupCallbacks::default();
+    let managed_group_config = ManagedGroupConfig::new(
+        handshake_message_format,
+        update_policy,
+        0,
+        0,
+        true, // use_ratchet_tree_extension
+        callbacks,
+    );
+    let setup = ManagedTestSetup::new(managed_group_config, 8);
+
+    let group_id = setup
+        .create_random_group(8, Ciphersuite::default())
+        .unwrap();
+
+    let mut groups = setup.groups.borrow_mut();
+    let group = groups.get_mut(&group_id).unwrap();
+
+    // Get the id of the member at index 0
+    let (_, remover_id) = group
+        .members
+        .iter()
+        .find(|(index, _)| *index == 0)
+        .unwrap()
+        .clone();
+
+    // Remove the rightmost 2 members in the tree
+    setup
+        .remove_clients_by_index(ActionType::Commit, group, &remover_id, &[6, 7])
+        .expect("error while removing members from group");
+
+    // Test if the tree was truncated. The tree's size should be ((number of
+    // members) * 2) - 1, i.e. 11 with 6 members.
+    assert_eq!(group.public_tree.len(), 11)
```

Comment:
We also need to test the other case where no truncation happens, for example removing only one member in this scenario.

Reply:
Wouldn't this have to be a test for the `trim` function we use as part of the `remove` process? As a consumer of the `trim` function, I rely on it to do what it says.

Don't get me wrong, I'm happy to implement another test, either for the `remove` operation, or for the `trim` function. I'm just trying to understand what's the right approach here.

Reply:
It doesn't hurt to have tests for _both_, of course, but shouldn't the general approach be that I don't have to test for correct behaviour of all underlying functions I'm using?

Reply:
The code you add is a
```
if remove {
 trim
}
```
So the tests to cover this must cover
* `if remove` is `true`
  * `trim` has an effect
  * `trim` has no effect
* `if remove` is `false`

That `trim` itself isn't actually tested is another issue. But that's why it's even more important to test properly when it's used.

Reply:
Ah, then I misunderstood. I thought you wanted to test if trim has an effect if we remove one member, but there's nothing to trim. That sounds sensible.

Reply:
Although I'm not sure how to test this actually, because trim will really only have an effect if someone is removed at the right-most leaf of the tree. Consequently, in all other cases calling trim will have no effect, so we can't effectively test whether trim is run or not. We could of course remove the `if` statement as a consequence, but then again the spec says to only trim after removal 🤷🏻

Reply:
> in all other cases calling trim will have no effect

there you have your options for test cases